clients/client-python/tests/unittests/dto/rel/test_partitioning_serdes.py

```diff
@@ -0,0 +1,569 @@
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import json
+import unittest
+from enum import Enum
+from unittest.mock import patch
+
+from gravitino.api.types.types import Types
+from gravitino.dto.rel.expressions.field_reference_dto import FieldReferenceDTO
+from gravitino.dto.rel.expressions.literal_dto import LiteralDTO
+from gravitino.dto.rel.partitioning.day_partitioning_dto import DayPartitioningDTO
+from gravitino.dto.rel.partitioning.function_partitioning_dto import (
+    FunctionPartitioningDTO,
+)
+from gravitino.dto.rel.partitioning.hour_partitioning_dto import HourPartitioningDTO
+from gravitino.dto.rel.partitioning.identity_partitioning_dto import (
+    IdentityPartitioningDTO,
+)
+from gravitino.dto.rel.partitioning.json_serdes.partitioning_serdes import (
+    PartitioningSerdes,
+)
+from gravitino.dto.rel.partitioning.month_partitioning_dto import MonthPartitioningDTO
+from gravitino.dto.rel.partitioning.partitioning import Partitioning
+from gravitino.dto.rel.partitioning.year_partitioning_dto import YearPartitioningDTO
+from gravitino.exceptions.base import IllegalArgumentException
+
+
+class MockPartitionStrategy(str, Enum):
+    INVALID_STRATEGY = "invalid_partitioning_strategy"
+
+
+class TestPartitioningSerdes(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.field_name = [f"dummy_field_{i}" for i in range(1)]
+        cls.single_field_partitioning_dtos = {
+            Partitioning.Strategy.IDENTITY: IdentityPartitioningDTO(*cls.field_name),
+            Partitioning.Strategy.YEAR: YearPartitioningDTO(*cls.field_name),
+            Partitioning.Strategy.MONTH: MonthPartitioningDTO(*cls.field_name),
+            Partitioning.Strategy.DAY: DayPartitioningDTO(*cls.field_name),
+            Partitioning.Strategy.HOUR: HourPartitioningDTO(*cls.field_name),
+        }
+
+    def test_serialize_invalid_strategy(self):
+        mock_dto = IdentityPartitioningDTO(*self.field_name)
+        with patch.object(
+            mock_dto, "strategy", return_value=MockPartitionStrategy.INVALID_STRATEGY
+        ):
+            self.assertRaisesRegex(
+                IOError,
+                "Unknown partitioning strategy",
+                PartitioningSerdes.serialize,
+                mock_dto,
+            )
+
+    def test_deserialize_invalid_json(self):
+        invalid_partitioning_data = (None, "invalid_data", {})
```

Comment:
can we make `invalid_partitioning_data` a JSON string, and use json.load to load it ?
Just like what `test_deserialize_identity_partitioning_dto` did.